### PR TITLE
Adding 'name' parameter to POST /templates/repo

### PR DESCRIPTION
**Pull request by jcockbain96**
Adds an extra parameter in the request body of `POST /templates/repo` for repo name (`name`), this will take precedence over the name read from the `templates.json` in the repo. 

Also makes `name` and `description` both optional parameters for this API, with the user description taking precedence too.

Updates templates.service to match new template descriptions, which had been changed without updating the tests.

Resolves https://github.com/eclipse/codewind/issues/586.

Latest Commit - adds source to each template endpoint (if a repo name exists) - #608, and updates the default repo list to have a name for the Kabenero repo - #609.

Signed-off-by: James Cockbain <james.cockbain@ibm.com>

### CODE REVIEW

File: src/pfe/portal/modules/Templates.js [hunk 1]
```diff
@@ -325,7 +326,8 @@ async function readRepoTemplatesJSON(repository) {
   try {
     const templateDetails = JSON.parse(res.body);
     for (const prop of ['name', 'description']) {
-      if (templateDetails.hasOwnProperty(prop)) {
+      // Only set the name or description of the repo if not given by the user
+      if (templateDetails.hasOwnProperty(prop) && !repository[prop]) {
```
Comment (rwalle61) [inline src/pfe/portal/modules/Templates.js]: this should work, though performance could be improved if the `!repository[prop]` check were done before GETting the templates.json, to skip that request when name and desc are already set?

File: test/src/unit/template.test.js [hunk 2]
```diff
@@ -399,10 +400,11 @@ describe('Templates.js', function() {
             it('succeeds', async function() {
```
Comment (rwalle61) [inline test/src/unit/template.test.js]: the `describe`s now should have `<validName>` or something to similar effect

File: test/src/unit/template.test.js [hunk 3]
```diff
@@ -412,6 +412,20 @@ describe('Templates.js', function() {
                 });
             });
         });
+        describe('(<validUrl>, <NoDesc> <NoName>)', function() {
+            it('succeeds, and gets the name and description from templates.json', async function() {
+                const url = 'https://raw.githubusercontent.com/kabanero-io/codewind-templates/master/devfiles/index.json';
+                const func = () => templateController.addRepository(url);
+                await (func().should.not.be.rejected);
+                templateController.repositoryList.should.deep.include({
```
Comment (rwalle61) [inline test/src/unit/template.test.js]: can this instead use an object from `templates.service.js`? (To reduce maintenance)

File: src/pfe/portal/modules/Templates.js (hunk 1 above)
Comment (jcockbain96) [reply src/pfe/portal/modules/Templates.js]: What if only one is set?

Comment (rwalle61) [reply src/pfe/portal/modules/Templates.js]: in that case we still need to get the one that isn't set by the user, right?

Comment (jcockbain96) [reply src/pfe/portal/modules/Templates.js]: Sorry, yeah understood what you meant so deleted the comment

**Comment by rajivnathan** [comment]
@jcockbain96  The build for this PR seems to be green. Will this change fix the error I'm seeing in my PR build?

```
15:08:17    1) Batch enabling repositories
15:08:17         1 repo
15:08:17           returns 207 and sub-status 200 for each subrequest when batch disabling 1 repos:
15:08:17       Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/genie.codewind/jenkins-agent/workspace/Codewind_codewind_PR-604/test/src/API/templates/enableRepo.test.js)
```

**Comment by jcockbain96** [comment]
@rajivnathan I get that one passing on master so I don't think so - were you referring to https://github.com/eclipse/codewind/pull/604? Is green now so may just be it's sometimes hitting that timeout? We can up it if so.

**Pull request merged**